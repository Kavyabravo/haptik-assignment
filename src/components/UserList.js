import React from "react";

function User(props) {
  return (
    <React.Fragment>
      {props.userList.map((user) => {
        return (
          <div
            className={`row user-list`}
            key={`${user.id}-li-user`}
            onClick={() => props.handleSelectFriend(user)}
          >
            {user.name}
          </div>
        );
      })}
    </React.Fragment>
  );
}
export default User;
