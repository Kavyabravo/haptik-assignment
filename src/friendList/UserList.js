function User(props) {
    return (
        <div className="row">
          {props.userList.map(user => {
              return <li className="user-list"
              key={`${user.id}-li-user`} onClick={()=>props.addFriend(user)}>
                      {user.name}
                     </li>
          })}
          
      </div>
      )
} 
export default User;