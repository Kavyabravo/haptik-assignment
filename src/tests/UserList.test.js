import React from "react";
import renderer from "react-test-renderer";
import UserList from "../components/UserList";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { cleanup } from "@testing-library/react";
// import Link from '../Link.react';

afterEach(cleanup);
configure({ adapter: new Adapter() });
test("User List Snapshot", () => {
  const component = renderer.create(
    <UserList userList={[{ id: "1", name: "kav", isFav: true }]}></UserList>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Number of div created in Userlist component", () => {
  const component = shallow(
    <UserList
      userList={[
        { id: "1", name: "kav", isFav: true },
        { id: "2", name: "kavya", isFav: true },
      ]}
    ></UserList>
  );
  expect(component.find("div").length).toEqual(2);
});
test("UserList selects new friend on a click event", () => {
  const userList = [
    { id: "1", name: "kav", isFav: true },
    { id: "2", name: "kavya", isFav: true },
  ];
  let newFriend = {};
  const handleSelectFriend = jest.fn().mockImplementation(() => {
    newFriend = userList[0];
  });
  // Render a checkbox with label in the document
  const checkbox = shallow(
    <UserList userList={userList} handleSelectFriend={handleSelectFriend} />
  );

  expect(newFriend).toEqual({});

  checkbox.find("div").at(0).simulate("click");

  expect(newFriend).toEqual(userList[0]);
});
