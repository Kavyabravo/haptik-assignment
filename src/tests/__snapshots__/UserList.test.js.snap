// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User List Snapshot 1`] = `
<div
  className="row user-list"
  onClick={[Function]}
>
  kav
</div>
`;
